nx/lib/nx.ex

```diff
@@ -7919,6 +7921,101 @@ defmodule Nx do
     impl!(a, b).triangular_solve(%{b | type: output_type}, a, b, [])
   end
 
+  @doc """
+  Invert a square 2-D tensor.
+
+  For non-square tensors, use `Nx.svd/2` for pseudo-inverse calculations.
+
+  ## Examples
+
+      iex> a = Nx.tensor([[1, 0, 0, 0], [2, 1, 0, 0], [1, 0, 1, 0], [1, 1, 1, 1]])
+      iex> a_inv = Nx.invert(a)
+      #Nx.Tensor<
+        f32[4][4]
+        [
+          [1.0, 0.0, 0.0, 0.0],
+          [-2.0, 1.0, 0.0, 0.0],
+          [-1.0, 0.0, 1.0, 0.0],
+          [2.0, -1.0, -1.0, 1.0]
+        ]
+      >
+      iex> Nx.dot(a, a_inv)
+      #Nx.Tensor<
+        f32[4][4]
+        [
+          [1.0, 0.0, 0.0, 0.0],
+          [0.0, 1.0, 0.0, 0.0],
+          [0.0, 0.0, 1.0, 0.0],
+          [0.0, 0.0, 0.0, 1.0]
+        ]
+      >
+      iex> Nx.dot(a_inv, a)
+      #Nx.Tensor<
+        f32[4][4]
+        [
+          [1.0, 0.0, 0.0, 0.0],
+          [0.0, 1.0, 0.0, 0.0],
+          [0.0, 0.0, 1.0, 0.0],
+          [0.0, 0.0, 0.0, 1.0]
+        ]
+      >
+
+  ### Error cases
+
+      iex> Nx.invert(Nx.tensor([[3, 0, 0, 0], [2, 1, 0, 0]]))
+      ** (ArgumentError) can only invert square matrices, got: {2, 4}
+
+      iex> Nx.invert(Nx.tensor([[0, 0, 0, 0], [0, 0, 0, 0], [0, 0, 0, 0], [1, 1, 1, 1]]))
+      ** (ArgumentError) can't invert a singular matrix
+
+  """
+  @doc type: :linalg
+  def invert(tensor) do
+    %T{shape: {m, n}} = tensor = tensor!(tensor)
+
+    if m != n,
+      do:
+        raise(
+          ArgumentError,
+          "can only invert square matrices, got: {#{m}, #{n}}"
+        )
```

Comment:
This is unnecessarily verbose in this format. Maybe it should be a regular do block?

```suggestion
    if m != n do
      raise ArgumentError,
              "can only invert square matrices, got: {#{m}, #{n}}"
    end
```